File: app/src/main/java/com/zhuazhu/frame/MainActivity.java
package com.zhuazhu.frame;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import com.zhuazhu.annotation.LayoutId;
import com.zhuazhu.frame.data.HttpHelper;
import com.zhuazhu.frame.di.component.DaggerMainComponent;
import com.zhuazhu.frame.di.module.MainModule;
import javax.inject.Inject;
import mejust.frame.mvp.view.BaseActivity;
import mejust.frame.widget.TitleBarOptions;

@LayoutId(R.layout.activity_main)
public class MainActivity extends BaseActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        DaggerMainComponent.builder()
                .appComponent(FrameApplication.getAppComponent())
                .mainModule(new MainModule())
                .build()
                .inject(this);
        TitleBarOptions options = new TitleBarOptions();
        options.setTitleString("Hello World");

    }

    @Inject
    HttpHelper httpHelper;

    @Override
    public void onClick(View v) {
        Intent intent = new Intent();
        switch (v.getId()) {
            case R.id.image:
                intent.setClass(this, ImageActivity.class);
                startActivity(intent);
                break;
        }
    }
}

File: app/src/main/java/com/zhuazhu/frame/mvp/main/view/MainActivity.kt
package com.zhuazhu.frame.mvp.main.view

import android.Manifest
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import com.zhuazhu.frame.R.layout
import com.zhuazhu.frame.mvp.common.CommonActivity
import com.zhuazhu.frame.mvp.http.view.HttpActivity
import com.zhuazhu.frame.mvp.image.ImageActivity
import com.zhuazhu.frame.mvp.main.model.PageItem
import kotlinx.android.synthetic.main.activity_main.recycler_main
import mejust.frame.data.annotation.Title
import permissions.dispatcher.NeedsPermission
import permissions.dispatcher.RuntimePermissions

@RuntimePermissions
class MainActivity : AppCompatActivity() {

    private val pageItems = listOf(
        PageItem("image", ImageActivity::class.java),
        PageItem("http", HttpActivity::class.java),
        PageItem("common", CommonActivity::class.java)
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout.activity_main)
        recycler_main.apply {
            layoutManager = LinearLayoutManager(this@MainActivity)
            setHasFixedSize(true)
            addItemDecoration(
                DividerItemDecoration(
                    this@MainActivity,
                    LinearLayoutManager.VERTICAL
                )
            )
            adapter = MainAdapter(pageItems)
        }
        requestPermissionWithPermissionCheck()
    }

    @NeedsPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
    fun requestPermission() {

    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        onRequestPermissionsResult(requestCode, grantResults)
    }
}
